### PR TITLE
Change query to use a substitute for 'exclude'

### PR DESCRIPTION
**Pull request by stephherbers**
## Summary

Replaces the "exclude" filtering that was working, but now is taking way too long to run.

## Product Description

In all previous testing, the `exclude` was working, but now it takes way too long if it ever completes. Interesting using the `~Q` syntax like: `SQLUserRole.objects.filter(~Q(rolepermission__permission_fk_id=permission.id))` also takes too long to complete. I had to do a roundabout way where I fetch all roles and then subtract from that set the elements that do have the permission. I have verified via that shell that it returns the correct number of roles and also that those roles do not contain the new permission and quickly. 

## Safety Assurance

- [X] Risk label is set correctly
- [X] All migrations are backwards compatible and won't block deploy
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [X] If QA is part of the safety story, the "Awaiting QA" label is used
- [X] I have confidence that this PR will not introduce a regression for the reasons below

### QA Plan

I am not requesting QA

### Safety story

The previous filter is currently not completing within an hour of running it so is currently non-functional.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/users/management/commands/add_download_reports_permission.py [hunk 1]
```diff
@@ -9,7 +9,9 @@ class Command(BaseCommand):
     def handle(self, **options):
         permission, created = SQLPermission.objects.get_or_create(value='download_reports')
         num_roles_modified = 0
-        for role in SQLUserRole.objects.exclude(rolepermission__permission_fk_id=permission.id).iterator():
+        all_roles = SQLUserRole.objects.all()
+        roles_with_new_permission = SQLUserRole.objects.filter(rolepermission__permission_fk_id=permission.id)
+        for role in all_roles.difference(roles_with_new_permission).iterator():
```
Comment (snopoke) [inline corehq/apps/users/management/commands/add_download_reports_permission.py:14]:
```suggestion
        all_role_ids = set(SQLUserRole.objects.all().values_list("id", flat=True))
        role_ids_with_new_permission = set(SQLUserRole.objects.filter(rolepermission__permission_fk_id=permission.id).value_list("id", flat=True))
        difference = all_role_ids.difference(role_ids_with_new_permission)
        for chunk in chunked(difference):
```

**Review by snopoke — CHANGES_REQUESTED** [review]
I think you should only select the IDs and then re-fetch the roles you want to migrate in chunks.

I see the exclude query looks like this:

```sql
SELECT ... FROM "users_userrole" WHERE NOT ("users_userrole"."id" IN (SELECT U1."role_id" FROM "users_rolepermission" U1 WHERE U1."permission_fk_id" = %s))
```
Which my be OK for a small number of IDs in the list but if it's matching most of the table it will do a full table scan.

File: corehq/apps/users/management/commands/add_download_reports_permission.py (hunk 1 above)
Comment (stephherbers) [reply corehq/apps/users/management/commands/add_download_reports_permission.py]: I made change here for chunks of 50: 00968ac

Comment (stephherbers) [reply corehq/apps/users/management/commands/add_download_reports_permission.py]: @snopoke does this change look like what you had in mind?

**Review by esoergel — APPROVED** [review]

**Review by orangejenny — APPROVED** [review]

**Comment by stephherbers** [comment]
Merging with 2 thumbs up after implementing the clear changes requested and verifying the functionality with testing.

**Pull request merged**